### DISCUSSION TITLE
Doc as parser errors

### DISCUSSION THREAD

**Discussion by ozwaldorf** [pacak/bpaf#368] [Ideas]
Using doc as the parsing error for from_str currently doesn't allow you to actually render the doc with the terminal directed output. Even using `doc.monochrome(true)` and a string error type, output is always unstylized.

It would be awesome to expose a render_terminal method, that matches the styling of helptext and builtin errors to create a consistent cli experience

**Comment by pacak** [comment]
Yea, I think I'll do that in a day or two. Same release will include `fallback_to_usage` for derive macro (already implemented).

**Reply by ozwaldorf** [reply]
Amazing! Side comment; will fallback_to_usage work on both the top level command as well as subcommands?

**Reply by pacak** [reply]
> will fallback_to_usage work on both the top level command as well as subcommands?

Top level and subcommands, so `foo subcmd` where subcmd asks for fallback - prints it as well. You could always use it in combinatoric mode and this is just a change to the derive macro.

**Reply by ozwaldorf** [reply]
Saw the commit for fallback to usage, but I cant seem to add it to lutgen's top level command enum, always says the method isn't implemented when adding it here: https://github.com/ozwaldorf/lutgen-rs/blob/main/src/bin.rs#L508

```
E0599: no method named `fallback_to_usage` found for struct `ParseCommand` in the current scope
```

**Reply by pacak** [reply]
Can you push a branch with it added? Will take a look.

**Reply by ozwaldorf** [reply]
@pacak https://github.com/ozwaldorf/lutgen-rs/blob/fallback_to_usage/src/bin.rs
and failure shown in ci: https://github.com/ozwaldorf/lutgen-rs/runs/27834512467

**Reply by pacak** [reply]
Fixed in https://github.com/pacak/bpaf/pull/378
